apps/backend/src/factory/xlsx/fap.ts

```diff
@@ -180,46 +184,69 @@ export const collectFaplXLSXData = async (
     })
   );
 
+  const instrumentProposalsAnswers = await Promise.all(
+    instrumentsProposals.map((proposals) => {
+      return Promise.all(
+        proposals.map((proposal) =>
+          proposal
+            ? baseContext.queries.questionary.getQuestionarySteps(
+                user,
+                proposal.questionaryId
+              )
+            : null
+        )
+      );
+    })
+  );
+
   const out: FapXLSXData = [];
 
-  instruments.forEach((instrument, indx) => {
-    const proposals = instrumentsProposals[indx];
-    const proposalReviews = proposalsReviews[indx];
-    const proposalPrincipalInvestigators =
-      proposalsPrincipalInvestigators[indx];
-    const technicalReviews = proposalsTechnicalReviews[indx];
-    const fapProposals = instrumentsFapProposals[indx];
-    const fapMeetingDecisions = proposalsFapMeetingDecisions[indx];
-
-    const rows = proposals.map((proposal, pIndx) => {
-      const { firstname = '<missing>', lastname = '<missing>' } =
-        proposalPrincipalInvestigators[pIndx] ?? {};
-      const technicalReview = technicalReviews[pIndx];
-      const reviews = proposalReviews[pIndx];
-      const fapProposal = fapProposals?.[pIndx];
-      const fapMeetingDecision = fapMeetingDecisions[pIndx];
-
-      const proposalAverageScore = average(getGrades(reviews)) || 0;
-
-      return fapDataRow(
-        `${firstname} ${lastname}`,
-        proposalAverageScore,
-        instrument,
-        fapMeetingDecision,
-        proposal,
-        technicalReview,
-        fapProposal
+  await Promise.all(
+    instruments.map(async (instrument, indx) => {
+      const proposals = instrumentsProposals[indx];
+      const proposalReviews = proposalsReviews[indx];
+      const proposalPrincipalInvestigators =
+        proposalsPrincipalInvestigators[indx];
+      const technicalReviews = proposalsTechnicalReviews[indx];
+      const fapProposals = instrumentsFapProposals[indx];
+      const fapMeetingDecisions = proposalsFapMeetingDecisions[indx];
+      const proposalsAnswers = instrumentProposalsAnswers[indx];
+
+      const rows = await Promise.all(
+        proposals.map(async (proposal, pIndx) => {
```

Comment:
Not sure why do we need an async function here. I don't see anything asynchronous inside this function.

Reply:
I made it async as the STFC `fapDataRow` is asynchronous: https://github.com/UserOfficeProject/user-office-core/pull/492/commits/9f0e8477c9836201c3c00b625e6a1a5b57114bba#diff-0f8ddd45ec0c99053d9e83df998c997723ace6e88b91a18f19f20e1e28d4be3dR36

Reply:
I see now. Thank you for clarification.